app/lib/git_repository.rb

```diff
@@ -645,11 +671,14 @@ def directories_at_path(path)
   # It returns an array to ensure ordering, so that a directory
   # will always appear before any of the files or subdirectories
   # contained within it
-  def tree_at_path(path)
+  def tree_at_path(path, _finalize=true)
     result = entries_at_path(path).to_a
     result.select { |_, obj| obj.is_a? Repository::RevisionDirectory }.each do |dir_path, _|
-      result.push(*(tree_at_path(File.join(path, dir_path)).map { |sub_pth, obj| [File.join(dir_path, sub_pth), obj] }))
+      result.push(*(tree_at_path(File.join(path, dir_path), false).map do |sub_pth, obj|
+        [File.join(dir_path, sub_pth), obj]
+      end))
     end
+    return _add_push_time(result) if _finalize
```

Comment:
I'd like to see this line combined with the next. You can use a ternary if if you like.